### DISCUSSION TITLE
Proposal: Rename `class` to `trait`

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#6434] [Ideas]
This is a proposal to rename classes to traits (and thereby the `class`keyword to `trait`).

The rationale is small, but still existent: 

- The keyword `trait`is more common these days.
- It avoids confusion between Java classes and Flix type classes.

The downside is that it is mostly bike-shedding.

**Comment by jaschdoc** [comment]
I think this actually a worthwhile change. People used to Scala, Rust etc. will be immediately recognize the syntax, and allows us to have clear naming in documentation etc.

**Comment by stephentetley** [comment]
I also think this is a good change, especially for Magnus's second point.

**Comment by magnus-madsen** [comment]
We have decided to do this. Hence the discussion is now closed.
